tools/dependencies-report/src/main/java/org/logstash/dependencies/ReportGenerator.java

```diff
@@ -141,8 +141,16 @@ private void checkDependencyLicense(Map<String, LicenseUrlPair> licenseMapping,
         if (licenseMapping.containsKey(nameAndVersion)) {
             LicenseUrlPair pair = licenseMapping.get(nameAndVersion);
 
-            if (pair.url != null && !pair.url.equals("") &&
-               (acceptableLicenses.stream().anyMatch(pair.license::equalsIgnoreCase))) {
+            String[] dependencyLicenses = pair.license.split("\\|");
+            boolean hasAcceptableLicense = false;
+            for (int k = 0; k < dependencyLicenses.length && !hasAcceptableLicense; k++) {
+                if (pair.url != null && !pair.url.equals("") &&
```

Comment:
maybe it's worth adjusting the control flow here by moving this check to the upper level, and only perform the iteration over `dependencyLicenses` if `pair.url != null && !pair.url.equals("")`, otherwise leave `hasAcceptableLicense` as false. 
checking `pair.url != null && !pair.url.equals("")` on each iteration doesn't make much sense, even when putting performance argument aside.